### PR TITLE
:book: update docs and docs-gen according to hugo syntax

### PR DESCRIPTION
**Pull request by avinal**
## Summary

- [add cobra cli-doc-gen as third party](https://github.com/kcp-dev/kcp/pull/2120/commits/bf26ff3594e963955e87f72d3bc42eb6a8ba0812)
- [hack cli-docs generator to comply hugo syntax](https://github.com/kcp-dev/kcp/pull/2120/commits/16f27463be08e13b7674e5ddc2e2cb9a1d23a1bb)
  - Original code: https://github.com/spf13/cobra/blob/main/doc/md_docs.go
  - Hugo front matter: https://gohugo.io/content-management/front-matter/
- [update kcp docs according to hugo syntax](https://github.com/kcp-dev/kcp/pull/2120/commits/c92a66c4e8354193441fdab4db747a7daa9b9a7e)
  - remove cli documentation see https://github.com/kcp-dev/kcp/issues/2119#issuecomment-1265831212
  - add front matter to all docs
  - fix linting errors in markdown docs

Signed-off-by: Avinal Kumar <avinal@redhat.com>

## Test

Test deployment: https://avinal.space/kcp-dev.github.io/docs (might have lots of broken links, rest works)

### CODE REVIEW

File: hack/generate/cli-doc/gen-cli-doc.go [hunk 1]
```diff
@@ -14,16 +15,168 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// This program is a hacked version of the cobra cli-doc generator to generate
```
Comment (ncdc) [inline hack/generate/cli-doc/gen-cli-doc.go:18]: Please put this in `hack/third_party/github.com/spf13/cobra`. Thanks!

**Comment by avinal** [comment]
I essentially copied the whole cobra cli docs generation code and modified them to produce hugo compatible docs. Default options don't provide as much control over output. I hope this won't be considered over-engineered 😅 .

File: hack/generate/cli-doc/gen-cli-doc.go (hunk 1 above)
Comment (ncdc) [reply hack/generate/cli-doc/gen-cli-doc.go:18]: Would also be nice to split into 2 commits:
1. Copy in the cobra file unmodified
2. Modify the file

Comment (avinal) [reply hack/generate/cli-doc/gen-cli-doc.go:18]: Noted

**Comment by avinal** [comment]
Please suggest appropriate titles and descriptions, since I am still exploring kcp, I wasn't able to be much creative.

File: docs/concepts.md [hunk 2]
```diff
@@ -1,9 +1,17 @@
-# Terminology for control plane
+---
+title: "Terminology for control plane"
+linkTitle: "Terminology"
+weight: 1
+description: >
+  Terminology for control plane
```
Comment (sttts) [inline docs/concepts.md:6]: for kcp?

File: docs/developers/library-usage.md [hunk 3]
```diff
@@ -1,4 +1,10 @@
-# Using `kcp` as a library
+---
+title: "Using `kcp` as a library"
+linkTitle: "Library Usage"
+weight: 1
+description: >
+  How to use `kcp` as a library
```
Comment (sttts) [inline docs/developers/library-usage.md:6]: without backticks

File: docs/developers/releasing.md [hunk 4]
```diff
@@ -1,18 +1,24 @@
-# Publishing a new kcp release
-This document describes the processes to follow when creating and publishing a new kcp release.
+---
+title: "Publishing a new kcp release"
+linkTitle: "Publishing"
+weight: 1
+description: >
+  This document describes the processes to follow when creating and publishing a new kcp release.
```
Comment (sttts) [inline docs/developers/releasing.md:6]: kcp release process ?

**Pull request merged**